File: clients/js/src/requests/move_names.ts
import { BaseRequest } from './base';

export type MoveNameRequest = BaseRequest<MoveNameRequestType, MoveNameRequestAttributes>;

enum MoveNameRequestType {
    move_names = 'move_names',
}

export interface MoveNameRequestAttributes {
    name: string;
}

export const MoveNameRequestDefault = (attributes: MoveNameRequestAttributes): MoveNameRequest => {
    return {
        data: {
            type: MoveNameRequestType.move_names,
            attributes,
        },
    };
};

File: pkmnapi-api/src/requests/move_names.rs
use rocket_okapi::JsonSchema;
use serde::Deserialize;

use crate::requests::base::BaseRequest;

pub type MoveNameRequest = BaseRequest<MoveNameRequestType, MoveNameRequestAttributes>;

impl MoveNameRequest {
    pub fn get_name(&self) -> &String {
        &self.data.attributes.name
    }
}

#[derive(Debug, Deserialize, JsonSchema)]
#[allow(non_camel_case_types)]
pub enum MoveNameRequestType {
    move_names,
}

#[derive(Debug, Deserialize, JsonSchema)]
pub struct MoveNameRequestAttributes {
    pub name: String,
}
